Adds a note that source maps are not uploaded for debug mode in React Native / Expo (#11478)
getsentry/sentry-docs: docs/platforms/react-native/sourcemaps/index.mdx

    uploadNativeSymbols = shouldSentryAutoUpload()
}
```

> ### Debug and Simulator Builds
> 
> Note that uploading source maps for debug or simulator builds is disabled because symbolication is handled directly by the Metro development server. In a development environment, Metro automatically provides the symbolication needed to translate stack traces into readable source code. As a result, there is no need to upload source maps for these types of builds.

Reviewer: Release with simulator target will auto upload the source maps.

```suggestion
### Debug Builds

Note that automatic uploading of source maps for debug builds is disabled because symbolication is handled directly by the Metro development server. In a development environment, Metro automatically provides the symbolication needed to translate stack traces into readable source code. As a result, there is no need to upload source maps for these types of builds.
```
